src/box/iproto.cc

```diff
@@ -1459,12 +1677,40 @@ tx_accept_wpos(struct iproto_connection *con, const struct iproto_wpos *wpos)
 	}
 }
 
+/**
+ * Since the processing of requests within a transaction
+ * for a stream can occur in different fibers, we store
+ * a pointer to transaction in the stream structure.
+ * Check if message belongs to stream and there is active
+ * transaction for this stream. In case it is so, sets this\
+ * transaction for current fiber.
+ */
+static inline struct txn *
+tx_prepare_transaction_for_request(struct iproto_msg *msg)
+{
+	struct txn *txn = NULL;
+	if (msg != NULL && msg->stream != NULL) {
+		txn = msg->stream->txn;
+		/*
+		 * By default for all iproto_messages this flag sets to true.
+		 * But if there is active transaction for this stream and
+		 * message is processed within this transaction, then we
+		 * must change it to false.
+		 */
```

Comment:
Please write in comment what you want to achieve by setting this flag.

Reply:
Add comment